bundles/org.eclipse.e4.ui.progress/src/org/eclipse/e4/ui/progress/UIJob.java

```diff
@@ -48,15 +48,27 @@ public UIJob(String name) {
 	/**
 	 * Create a new instance of the receiver with the supplied Display.
 	 *
-	 * @param jobDisplay
-	 *            the display
-	 * @param name
-	 *            the job name
+	 * @param jobDisplay the display
+	 * @param name       the job name
 	 * @see Job
+	 * @deprecated use {@link #UIJob(UISynchronize, String)} to supply alternative
```

Comment:
How to get a `UISynchronize` from a Display? Would be good to mention that and suggest a blind replacement customers can use without wondering about anything else.

Reply:
> How to get a `UISynchronize` from a Display?

You don't get it from a display, you let inject it into your E4 part (the `UISynchronize`), see for example:

- https://www.vogella.com/tutorials/EclipseJobs/article.html#using-dependency-injection-and-uisynchronize

Reply:
Please explain that in Javadoc, or link to some other existing Javadoc that explains that.

Reply:
The fact that migration path are poorly documented is 1 reason why e4 is still not adopted even in the SDK. I would rather use the current interest as an opportunity to really make things better and easier for everyone to adopt e4 than to just repeat things that haven't been much successful. We need to raise the quality bar on documentation, and this means having good javadoc in the code more than referencing outdated wikis on PRs.

Reply:
i am also interested in a non e4 replacement.

Reply:
> **We** need to raise the quality bar on documentation, and this means having good javadoc in the code more than referencing outdated wikis on PRs.

Yep, **we** should do that, bit I can't afford handling the technical dept of 14 years platform, so the best opportunity here is to step up and start with it right now, but this should be handled in a dedicated change.

So let me know if actually making adoption of E4 possible and useful and accepting that an unknown class can't tell the whole story of E4 or I if I should simply revert, keep the old useless code and AP in and just fix the NPE to the bare minimum here...

Reply:
> So let me know if actually making adoption of E4 possible and useful

Yes, but deprecating for removal without good documentation is not. So maybe just keep the constructor based on Display at the moment without deprecating it.

Reply:
This code is effectively not used (even not internally) and gives false impression here, because it has non effect atm... so actually I would remove the constructor completely here as it was broken from day-1 here... I'm also not sure what esle schould be "better" documented here as that we will remove it and there is an (e4 compatible) alternative?

Reply:
> This code is effectively not used (even not internally)

That you cannot tell for sure when it's API.

> I'm also not sure what esle schould be "better" documented here as that we will remove it and there is an (e4 compatible) alternative?

As mentioned earlier, the expectation when documenting a deprecation is that people can read the deprecation notice from the javadoc and quickly enough know how to replace their code using the deprecated method by something else that is not deprecated. This is not about just telling "use that instead", but when the difference is non trivial, more "from your code, do this to get an instance of that and then apply this other method".

Reply:
Yeah forget about that stupid deprecation, one should just not try to deprecate anything and then we even don't need a simpler "removal process at all, so all win-win...

Reply:
>  one should just not try to deprecate anything 

That's not the goal; but it's the drawback of declaring something as API. For a Platform that claims sustainability like Eclipse Platform, creating an API means that the API will be maintained as working without regression, will not be removed without a notice and that it's removal will be accompanied with clear steps on how to migrate. Don't do it well and the Platform breaks clients too strongly, extensions don't work, people move to other platforms, ecosystem dies, community dies, technology dies... That is the real cost and risk with addition APIs, and this cost is something you easily forget when you create an API, as the project pays it later, APIs contract do reduce agility and capability to innovate. But APIs are a necessary evil. I hope that those case-studies help you understand why I'm always defensive when it comes to API addition. There will always be a payback, and often we get the impression having an API was overall not profitable and world would have been better without it.